packages/core/src/combinator/timeslice.js

```diff
@@ -3,111 +3,102 @@
 /** @author John Hann */
 
 import Pipe from '../sink/Pipe'
-import { disposeAll } from '@most/disposable'
+import { disposeBoth } from '@most/disposable'
 import { join } from './chain'
+import { merge } from './merge'
+import { never } from '../source/never'
+
+const timeBounds = (min, minSignal, maxSignal) =>
+  ({ min, minSignal, maxSignal })
+
+const minBounds = (minSignal) =>
+  timeBounds(Infinity, minSignal, never())
```

Comment:
`Infinity` here is correct, but probably not obvious at first.  If you think about what a minimum bound _does_, then it's more obvious: we want to drop all events before we see the initial event from minSignal.  By starting with `Infinity`, we're saying that we want to discard all events before `Infinity`.  The first event in minSignal provides a new min (presumably earlier than `Infinity`!).